### ISSUE TITLE
Forkchoice crashed

### ISSUE THREAD

**Issue by mpetrunic** [ChainSafe/lodestar#1314]
**Describe the bug**

While syncing near head of altona, forkchoice crashed.

![image](https://user-images.githubusercontent.com/8836210/89230769-47acfb00-d5e4-11ea-8748-7e2731cb7f4f.png)

**Comment by mpetrunic** [comment]
Full log:
```
2020-08-03 23:48:48  [REGULAR_SYNC_V2]  info: Received blocks for range blockCount=5, startSlot=254561, endSlot=254575
2020-08-03 23:48:48  [REGULAR_SYNC_V2]  info: Fetching block chunks validPeerCount=5
2020-08-03 23:48:48  [NETWORK]       verbose: sending beacon_blocks_by_range request to 16Uiu2HAmDUr2NS194XDgZ2rwHm7CyzR29j2si3bGapv4pT4oyLZy requestId=95e901eb4ed457ef, encoding=ssz_snappy
2020-08-03 23:48:48  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xa7f9dae98b90df3860423ed6cdabc3edabe44bf4715b9110b31bf89a17a426ff, slot=251200, epoch=7850
2020-08-03 23:48:48  [NETWORK]       verbose: got stream to 16Uiu2HAmDUr2NS194XDgZ2rwHm7CyzR29j2si3bGapv4pT4oyLZy requestId=95e901eb4ed457ef, encoding=ssz_snappy
2020-08-03 23:48:49  [CHAIN]            info: Processed new chain head newChainHeadRoot=0x9b8faebd2d3b9f4aa17741e97678a4a4261bceba210f6005eab51dbce58c08bc, slot=251201, epoch=7850
2020-08-03 23:48:49  [NETWORK]       verbose: receive beacon_blocks_by_range response with 3 chunks from 16Uiu2HAmDUr2NS194XDgZ2rwHm7CyzR29j2si3bGapv4pT4oyLZy requestId=95e901eb4ed457ef, encoding=ssz_snappy, body={"startSlot":"254577","count":"21","step":"1"}
2020-08-03 23:48:49  [REGULAR_SYNC_V2]  info: Received blocks for range blockCount=3, startSlot=254578, endSlot=254588
2020-08-03 23:48:49  [REGULAR_SYNC_V2]  info: Fetching block chunks validPeerCount=5
2020-08-03 23:48:49  [NETWORK]       verbose: sending beacon_blocks_by_range request to 16Uiu2HAmKvyXP7YTcKfyKHfssLVQDKbCSEMSnf6RnAbwynuBmkE5 requestId=34ee47b9445fc855, encoding=ssz_snappy
2020-08-03 23:48:50  [CHAIN]            info: Processed new chain head newChainHeadRoot=0x72481f3599e5a31d3578672a8a4e14dca334301034ede8ce8579b8d580c47901, slot=251206, epoch=7850
2020-08-03 23:48:50  [NETWORK]       verbose: got stream to 16Uiu2HAmKvyXP7YTcKfyKHfssLVQDKbCSEMSnf6RnAbwynuBmkE5 requestId=34ee47b9445fc855, encoding=ssz_snappy
2020-08-03 23:48:50  [NETWORK]       verbose: receive ping request from 16Uiu2HAkx9FCJQEuL6UaWQromgLEPntNQBwYsK441oTwP7g3PwAV requestId=dfcfe0a6dc6647d6, encoding=ssz_snappy
2020-08-03 23:48:50  [NETWORK]       verbose: Sent response for request dfcfe0a6dc6647d6 
2020-08-03 23:48:50  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xb145815a6591c1f3dc825ea8020e76e3d8b6d6ca55161b35f5ccb95047e911b2, slot=251211, epoch=7850
2020-08-03 23:48:50  [NETWORK]       verbose: receive beacon_blocks_by_range response with 4 chunks from 16Uiu2HAmKvyXP7YTcKfyKHfssLVQDKbCSEMSnf6RnAbwynuBmkE5 requestId=34ee47b9445fc855, encoding=ssz_snappy, body={"startSlot":"254598","count":"21","step":"1"}
2020-08-03 23:48:50  [REGULAR_SYNC_V2]  info: Received blocks for range blockCount=4, startSlot=254606, endSlot=254615
2020-08-03 23:48:50  [REGULAR_SYNC_V2]  info: Fetching block chunks validPeerCount=5
2020-08-03 23:48:50  [NETWORK]       verbose: sending beacon_blocks_by_range request to 16Uiu2HAmUsd7pEJkwtD7A7UUZi2tuL8kJU29giJwmyJn3wP1ErnD requestId=92162a3fd2b1500d, encoding=ssz_snappy
2020-08-03 23:48:50  [CHAIN]            info: Processed new chain head newChainHeadRoot=0x0878dafccdd2fb495f6cfc5cec9b70aac026a6dd207cd6706a8a32431e43d939, slot=251212, epoch=7850
2020-08-03 23:48:50  [NETWORK]       verbose: got stream to 16Uiu2HAmUsd7pEJkwtD7A7UUZi2tuL8kJU29giJwmyJn3wP1ErnD requestId=92162a3fd2b1500d, encoding=ssz_snappy
2020-08-03 23:48:51  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xb424929cef9e8ae213aa8e4a566c23986cde31c242d944fbf040770eeef13272, slot=251214, epoch=7850
2020-08-03 23:48:51  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xb95b5442b4cd583611a02903f4313f137facce48ef12c1e9d2f7e65c92977471, slot=251220, epoch=7850
2020-08-03 23:48:51  [NETWORK]       verbose: receive beacon_blocks_by_range response with 6 chunks from 16Uiu2HAmUsd7pEJkwtD7A7UUZi2tuL8kJU29giJwmyJn3wP1ErnD requestId=92162a3fd2b1500d, encoding=ssz_snappy, body={"startSlot":"254619","count":"21","step":"1"}
2020-08-03 23:48:51  [REGULAR_SYNC_V2]  info: Received blocks for range blockCount=6, startSlot=254619, endSlot=254639
2020-08-03 23:48:51  [REGULAR_SYNC_V2]  info: Fetching block chunks validPeerCount=5
2020-08-03 23:48:51  [NETWORK]       verbose: sending beacon_blocks_by_range request to 16Uiu2HAmUsd7pEJkwtD7A7UUZi2tuL8kJU29giJwmyJn3wP1ErnD requestId=96c2c565c67cea3c, encoding=ssz_snappy
2020-08-03 23:48:52  [NETWORK]       verbose: got stream to 16Uiu2HAmUsd7pEJkwtD7A7UUZi2tuL8kJU29giJwmyJn3wP1ErnD requestId=96c2c565c67cea3c, encoding=ssz_snappy
2020-08-03 23:48:52  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xbcd7d55f7b6f46f6097a7cb21f3005976331ed448e40b97bd7c1e1a03634154c, slot=251226, epoch=7850
2020-08-03 23:48:52  [NETWORK]       verbose: receive beacon_blocks_by_range response with 1 chunks from 16Uiu2HAmUsd7pEJkwtD7A7UUZi2tuL8kJU29giJwmyJn3wP1ErnD requestId=96c2c565c67cea3c, encoding=ssz_snappy, body={"startSlot":"254640","count":"21","step":"1"}
2020-08-03 23:48:52  [REGULAR_SYNC_V2]  info: Received blocks for range blockCount=1, startSlot=254642, endSlot=254642
2020-08-03 23:48:52  [REGULAR_SYNC_V2]  info: Fetching block chunks validPeerCount=5
2020-08-03 23:48:52  [NETWORK]       verbose: sending beacon_blocks_by_range request to 16Uiu2HAmTVrGFfE6sSw8aKmDFi1nyXa7qiRrAYEppL7DPJNxVCY2 requestId=04989dc4bd2fea31, encoding=ssz_snappy
2020-08-03 23:48:52  [CHAIN]            info: Processed new chain head newChainHeadRoot=0xf66fdff5bcf70cd69c5ed158902b01d7b2d0fa465b2d83e68a5c20b238e2d0d2, slot=251229, epoch=7850
2020-08-03 23:48:53  [CHAIN]            info: Processed new chain head newChainHeadRoot=0x15f1d404f813f081b358e16fb436cff7dab4642f500fb9862c7dd74f73b62deb, slot=251241, epoch=7851
2020-08-03 23:48:53  [NETWORK]       verbose: got stream to 16Uiu2HAmTVrGFfE6sSw8aKmDFi1nyXa7qiRrAYEppL7DPJNxVCY2 requestId=04989dc4bd2fea31, encoding=ssz_snappy
(node:19060) UnhandledPromiseRejectionWarning: RangeError: Maximum call stack size exceeded
    at Object.get [as GENESIS_EPOCH] (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/constants/index.js:13:22)
    at ArrayDagLMDGHOST.getJustifiedCheckpoint (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:617:45)
    at ArrayDagLMDGHOST.isCandidateForBestTarget (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:530:52)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:511:69)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:515:12)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:515:12)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:515:12)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:515:12)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:515:12)
    at ArrayDagLMDGHOST.propagateWeightChange (/home/mpetrunic/projects/eth2/lodestar/packages/lodestar/lib/chain/forkChoice/arrayDag/lmdGhost.js:515:12)
(node:19060) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:19060) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
2020-08-03 23:48:54  [NETWORK]       verbose: receive beacon_blocks_by_range response with 5 chunks from 16Uiu2HAmTVrGFfE6sSw8aKmDFi1nyXa7qiRrAYEppL7DPJNxVCY2 requestId=04989dc4bd2fea31, encoding=ssz_snappy, body={"startSlot":"254661","count":"21","step":"1"}

```

**Comment by mpetrunic** [comment]
@wemeetagain It could be that it occurs because it's calling this 5-6k times recursively and hit recursion depth limit.
https://stackoverflow.com/questions/20936486/node-js-maximum-call-stack-size-exceeded

We should probably wrap it in process.nextTick

**Comment by 3xtr4t3rr3str14l** [comment]
should we close this since altona is no longer relevant?  or is this issue happening in medalla as well?

**Comment by mpetrunic** [comment]
It's not really related to altona, this seems like some infinite (too much at least) recursion call

**Comment by wemeetagain** [comment]
We're working to refactor fork choice to remove recursion

**Comment by twoeths** [comment]
this will resolve the issue https://github.com/ChainSafe/lodestar/pull/1495

**Comment by mpetrunic** [comment]
resolved with #1495
